vms/platformvm/utxo/camino_locked.go

```diff
@@ -257,38 +259,62 @@ func (h *handler) Lock(
 	totalAmountBurned := uint64(0)
 
 	type lockedAndRemainedAmounts struct {
-		locked   uint64
-		remained uint64
+		lockedOrTransferred uint64
+		remained            uint64
 	}
-	type OwnerID struct {
-		owners   *secp256k1fx.OutputOwners
-		ownersID *ids.ID
+	type Owner struct {
+		secpOwners *secp256k1fx.OutputOwners
+		id         *ids.ID
 	}
 	type OwnerAmounts struct {
-		amounts map[ids.ID]lockedAndRemainedAmounts
-		owners  secp256k1fx.OutputOwners
+		amounts    map[ids.ID]lockedAndRemainedAmounts
+		secpOwners *secp256k1fx.OutputOwners
 	}
 	// Track the amount of transfers and their owners
 	// if appliedLockState == bond, then otherLockTxID is depositTxID and vice versa
 	// ownerID -> otherLockTxID -> AAAA
 	insAmounts := make(map[ids.ID]OwnerAmounts)
 
-	var toOwnerID *ids.ID
-	if to != nil && (appliedLockState == locked.StateUnlocked || appliedLockState == locked.StateDeposited) {
+	newOwner := Owner{}
+	if to != nil && appliedLockState == locked.StateUnlocked {
+		isNestedMsig, err := h.fx.IsNestedMultisig(to, utxoDB)
+		switch {
+		case err != nil:
+			err = fmt.Errorf("failed to check if to-owner is nested multisig owner: %w", err)
+		case isNestedMsig:
+			err = errNestedMultisigToOwner
+		}
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+
 		id, err := txs.GetOwnerID(to)
 		if err != nil {
+			err = fmt.Errorf("failed to get ownerID for to-owner: %w", err)
 			return nil, nil, nil, nil, err
 		}
-		toOwnerID = &id
+		newOwner = Owner{to, &id}
 	}
 
-	var changeOwnerID *ids.ID
+	changeOwner := Owner{}
 	if change != nil {
+		isNestedMsig, err := h.fx.IsNestedMultisig(change, utxoDB)
```

Comment:
Except one word in the error message and the usage of other variables this is the same code as in lines 280+. Maybe it's worth it to put that into a separate reusable function?